src/cr/cube/crunch_cube.py

```diff
@@ -1065,32 +1065,29 @@ def _inserted_dim_inds(self, transform_dims, axis):
         return np.array(inserted_inds[dim_ind] if len(inserted_inds) else [])
 
     def _insertions(self, result, dimension, dimension_index):
-        insertions = []
+        """Return list of (idx, sum) pairs representing subtotals.
 
-        for indices in dimension.hs_indices:
-            ind_subtotal_elements = np.array(indices['inds'])
-
-            if indices['anchor_ind'] == 'top':
-                ind_insertion = -1
-            elif indices['anchor_ind'] == 'bottom':
-                ind_insertion = result.shape[dimension_index] - 1
-            else:
-                ind_insertion = indices['anchor_ind']
-
-            ind = tuple(
-                [slice(None) for _ in range(dimension_index)] +
-                [ind_subtotal_elements]
-            )
-            axis = dimension_index
+        *idx* is the int offset at which to insert the ndarray subtotal
+        in *sum*.
+        """
 
-            # no indices are provided (should never get here)
-            if len(indices['inds']) == 0:
-                value = 0
-            else:
-                value = np.sum(result[ind], axis=axis)
-                insertions.append((ind_insertion, value))
+        def iter_insertions():
+            for anchor_idx, addend_idxs in dimension.hs_indices:
+                insertion_idx = (
+                    -1 if anchor_idx == 'top' else
+                    result.shape[dimension_index] - 1 if anchor_idx == 'bottom'
+                    else anchor_idx
+                )
+                addend_fancy_idx = tuple(
+                    [slice(None) for _ in range(dimension_index)] +
+                    [np.array(addend_idxs)]
+                )
+                yield (
+                    insertion_idx,
+                    np.sum(result[addend_fancy_idx], axis=dimension_index)
+                )
 
-        return insertions
+        return [insertion for insertion in iter_insertions()]
```

Comment:
I extracted the body of the code into the local method `iter_insertions()` to make this simpler to understand.